source/grammar/qasm3Parser.g4

```diff
@@ -203,7 +203,7 @@ arrayReferenceType: (READONLY | MUTABLE) ARRAY LBRACKET scalarType COMMA (expres
 
 designator: LBRACKET expression RBRACKET;
 
-defcalTarget: MEASURE | RESET | DELAY | Identifier;
+defcalTarget: MEASURE | RESET | DELAY LBRACKET Identifier RBRACKET | Identifier;
```

Comment:
Technical bikeshedding: I would still consider `delay` to be the target of the `defcal`, and the duration specifier to be a separate subrule of the statement.  Perhaps we could instead factor it as
```antlr4
defcalStatement:
    DEFCAL defcalTarget
    (LBRACKET defcalDuration RBRACKET)?
    (LPAREN defcalArgumentDefinitionList? RPAREN)?
    // ...
```
and leave `defcalTarget` alone?  Then in the AST visitor we'd error on any program that supplied the `defcalDuration` with an invalid target instead.  That leaves open the possibility of other identifiers being able to take a duration in their argument definition as well (not a language feature we're looking to add in the 3.0 release, but it's been discussed a fair amount in relation to a 3.1+ release).

The value of the rule `defcalDuration` to me depends on what we decide about the text of the spec first - the textual spec is our source of truth, and then this grammar is meant to be an indicative implementation of that.

(The reason I'm bringing it up when the outcome is identical right now is just because this ANTLR grammar is also meant to demonstrate how someone should _think_ about the parse tree of the language, as a counterpart to the written spec.)